inspirehep/modules/workflows/workflows/hep_ingestion.py

```diff
@@ -125,6 +130,11 @@ class mysource(HEPIngestion):
             with_author_keywords=True,
         ),
         filter_core_keywords,
+        guess_categories,
+        IF(is_experimental_paper, [
+            guess_experiments,
+        ]),
+        guess_keywords,
```

Comment:
I really like your latter proposal. IMHO we should have one workflow per document type. The fact that something comes from arXiv Vs. publisher Vs. user is of smaller importance given all the enhancements, checks, and involvements of cataloguers, the end product being always the same, i.e. an article.

So somehow I see these workflows as part of input branches of a tree going then to one output trunk.

On the other hand this might force the workflow to be unnecessary even more complex. Would be nice to once pict the overall steps involved...